src/java/org/apache/cassandra/tools/nodetool/CMSAdmin.java

```diff
@@ -269,4 +271,76 @@ public void execute(NodeProbe probe)
             probe.getCMSOperationsProxy().resumeDropAccordTable(tableId);
         }
     }
+
+    @Command(name = "dumpclustermetadata", description = "Dumps Cluster Metadata into a file")
+    public static class DumpClusterMetadata extends NodeTool.NodeToolCmd
+    {
+
+        @Option(title = "Epoch", name = { "-e", "--epoch" }, required = false,
+        description = "Epoch at which cluster metadata should be dumped")
+        private Long epoch;
+
+        @Option(title = "Transform Epoch", name = { "-te", "--transform-epoch" }, required = false,
+        description = "The epoch to which the cluster meta data should be transformed before dumping")
```

Comment:
maybe something like "force metadata to X epoch while dumping"

Reply:
done